examples/InlineDatum.purs

```diff
@@ -0,0 +1,154 @@
+import Contract.TxConstraints as Constraints
+import Contract.Utxos (UtxoM(UtxoM), utxosAt)
+import Contract.Value as Value
+import Data.BigInt as BigInt
+import Data.Map as Map
+
+main :: Effect Unit
+main = example testnetNamiConfig
+
+example :: ConfigParams () -> Effect Unit
+example cfg = launchAff_ do
+  runContract cfg do
+    logInfo' "Running Examples.InlineDatum"
+    validator <- checkDatumIsInlineScript
+    let vhash = validatorHash validator
+    logInfo' "Attempt to lock value"
+    txId <- payToCheckDatumIsInline vhash
+    awaitTxConfirmed txId
+    logInfo' "Tx submitted successfully, Try to spend locked values"
+    spendFromCheckDatumIsInline vhash validator txId
+  publishTestFeedback true
+
+plutusData :: PlutusData
+plutusData = Integer $ BigInt.fromInt 31415927
+
+payToCheckDatumIsInline :: ValidatorHash -> Contract () TransactionHash
+payToCheckDatumIsInline vhash = do
+  let
+    datum :: Datum
+    datum = Datum plutusData
+
+    constraints :: TxConstraints Unit Unit
+    constraints = Constraints.mustPayToScriptInlineDatum vhash datum
+      $ Value.lovelaceValueOf
+      $ BigInt.fromInt 2_000_000
+
+    lookups :: Lookups.ScriptLookups PlutusData
+    lookups = mempty
+
+  buildBalanceSignAndSubmitTx lookups constraints
+
+payToCheckDatumIsInlineWrong :: ValidatorHash -> Contract () TransactionHash
+payToCheckDatumIsInlineWrong vhash = do
+  let
+    datum :: Datum
+    datum = Datum plutusData
+
+    constraints :: TxConstraints Unit Unit
+    constraints = Constraints.mustPayToScript vhash datum
+      $ Value.lovelaceValueOf
+      $ BigInt.fromInt 2_000_000
+
+    lookups :: Lookups.ScriptLookups PlutusData
+    lookups = mempty
+
+  buildBalanceSignAndSubmitTx lookups constraints
+
+spendFromCheckDatumIsInline
+  :: ValidatorHash
+  -> Validator
+  -> TransactionHash
+  -> Contract () Unit
+spendFromCheckDatumIsInline vhash validator txId = do
+  let scriptAddress = scriptHashAddress vhash
+  UtxoM utxos <- fromMaybe (UtxoM Map.empty) <$> utxosAt scriptAddress
+  case fst <$> find hasTransactionId (Map.toUnfoldable utxos :: Array _) of
+    Just txInput -> do
+      let
+        redeemer :: Redeemer
+        redeemer = Redeemer plutusData
+
+        lookups :: Lookups.ScriptLookups PlutusData
+        lookups = Lookups.validator validator
+          <> Lookups.unspentOutputs utxos
+
+        constraints :: TxConstraints Unit Unit
+        constraints =
+          Constraints.mustSpendScriptOutput txInput redeemer
+
+      spendTxId <- buildBalanceSignAndSubmitTx lookups constraints
+      awaitTxConfirmed spendTxId
+      logInfo' "Successfully spent locked values."
+
+    _ ->
+      logInfo' $ "The id "
+        <> show txId
+        <> " does not have output locked at: "
+        <> show scriptAddress
+  where
+  hasTransactionId :: TransactionInput /\ _ -> Boolean
+  hasTransactionId (TransactionInput tx /\ _) =
+    tx.transactionId == txId
+
+buildBalanceSignAndSubmitTx
```

Comment:
Oh, we need to merge `develop` in soon where this helper is exported from `Examples.Helpers`